### DISCUSSION TITLE
How to check if queue exists?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [rabbitmq/rabbitmq-dotnet-client#1130] [Q&A] [answered]
Is there any way to check if queue exists?
For example,  a client app connects to RabbitMQ Server and it needs to check if queue exists before create it.
I do Declare , receive an Exception and the channel is closed.
Why no a simple method like Exists(QueueName) = true/false ?

**Comment by lukebakken** [comment, marked as answer]
> For example, a client app connects to RabbitMQ Server and it needs to check if queue exists before create it

Your application should just create the queue using identical creation parameters. If it already exists, nothing will happen.

If you need to check if a queue exists, use a passive declaration. See the `passive` parameter:

https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/RabbitMQ.Client/client/framing/QueueDeclare.cs#L52
